### PR TITLE
Add "list" action to storage resource access

### PR DESCRIPTION
**Pull request by edwardfoyle**
## Problem

The "read" action currently only grants "GetObject" access. Read should also include the ability to list objects (within the specific prefix).
**Issue number, if available:**

## Changes

Add the ListBucket action to the behavior of the "read" action. Additionally, two new actions, "get" and "list" are added so that behavior can be scoped to only get or list if desired. "read" is a convenience action that includes both "get" and "list".
**Corresponding docs PR, if applicable:**

## Validation

Updated unit tests and manually validated the generated policies
## Checklist

- [ ] If this PR includes a functional change to the runtime behavior of the code, I have added or updated automated test coverage for this change.
- [ ] If this PR requires a change to the [Project Architecture README](../PROJECT_ARCHITECTURE.md), I have included that update in this PR.
- [ ] If this PR requires a docs update, I have linked to that docs PR above.
- [ ] If this PR modifies E2E tests, makes changes to resource provisioning, or makes SDK calls, I have run the PR checks with the `run-e2e` label set.

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: packages/backend-storage/src/types.ts [hunk 1]
```diff
@@ -62,9 +62,22 @@ export type StorageAccessDefinition = {
   ownerPlaceholderSubstitution: string;
 };
 
-export type StorageAction = 'read' | 'write' | 'delete';
+/**
+ * Actions that can be granted to specific paths within the storage resource.
+ *
+ * 'read' grants both 'get' and 'list' actions.
```
Comment (sobolk) [inline packages/backend-storage/src/types.ts:68]: have we considered removing `read` ?

Comment (edwardfoyle) [reply packages/backend-storage/src/types.ts:68]: PMs didn't like that option :)

Comment (sobolk) [reply packages/backend-storage/src/types.ts:68]: kind of YAGNI violation.

**Review by sobolk — APPROVED** [review]

File: packages/backend-storage/src/storage_access_policy_factory.ts [hunk 2]
```diff
@@ -69,20 +70,48 @@ export class StorageAccessPolicyFactory {
...
+  get: ['s3:GetObject'],
+  list: ['s3:ListBucket'],
   write: ['s3:PutObject'],
   delete: ['s3:DeleteObject'],
 };
+
+/**
+ * Converts a prefix like /foo/bar/* into [foo/bar/, foo/bar/*]
+ * This is necessary to grant the ability to list all objects directly in "foo/bar" and all objects under "foo/bar"
+ */
```
Comment (0618) [inline packages/backend-storage/src/storage_access_policy_factory.ts:112]: If we have to remove the `/` prefix for the users, why don't we just have the user not use `/` at the beginning of the path? also, using `/` at the beginning means the root dir, but without `/` means the cwd, I think we mean cwd in our case, right?

**Review by 0618 — APPROVED** [review]

**Review by 0618 — APPROVED** [review]
Just left a question

File: packages/backend-storage/src/storage_access_policy_factory.ts (hunk 2 above)
Comment (edwardfoyle) [reply packages/backend-storage/src/storage_access_policy_factory.ts:112]: These are s3 prefixes, not local paths. The `/` at the beginning is meant to make it clear that all paths should start from the bucket root.

**Pull request merged**